src/storage/cache/rpc/client.rs

```diff
@@ -0,0 +1,665 @@
+        // Check and clean keeper timeout unused task
+        // The timeout data will be cleaned by the get function
+        // We don't need to clean the timeout data radically
+        let mut tickers = Box::pin(tokio::time::interval(
+            self.timeout_options.task_timeout.div_f32(2.0),
+        ));
+        loop {
+            // Clean timeout tasks, will block here
+            let recv_header_f = self.recv_header();
+            pin_mut!(recv_header_f);
+            let selector = ReceiveHeaderFuture::new(self, &mut tickers, &mut recv_header_f);
+            match selector.await {
+                Ok(header) => {
+                    // Try to receive the response from the server
+                    let header_seq = header.seq;
+                    debug!("{:?} Received keep alive response or other response.", self);
+                    let current_timestamp = self.clock_instant.elapsed().as_secs();
+                    self.received_keepalive_timestamp
+                        .store(current_timestamp, std::sync::atomic::Ordering::Release);
+                    // Update the received keep alive seq
+                    if let Ok(resp_type) = RespType::from_u8(header.op) {
+                        if let RespType::FileBlockResponse = resp_type {
+                            debug!("{:?} Received response header: {:?}", self, header);
+                            // Try to read to buffer
+                            match self.recv_len(header.len).await {
+                                Ok(()) => {}
+                                Err(err) => {
+                                    debug!(
+                                        "{:?} Failed to receive request header: {:?}",
+                                        self, err
+                                    );
+                                    break;
+                                }
+                            }
+
+                            // Take the packet task and recv the response
+                            let resp_buffer: &mut BytesMut = unsafe { &mut *self.resp_buf.get() };
+                            match self.packets_keeper.take_task(header_seq, resp_buffer).await {
+                                Ok(()) => {
+                                    debug!("{:?} Received response: {:?}", self, header_seq);
+                                }
+                                Err(err) => {
+                                    debug!("{:?} Failed to update task: {:?}", self, err);
+                                }
+                            }
+                        }
+                    } else {
+                        debug!("{:?} Invalid response type: {:?}", self, header.op);
+                        break;
+                    }
+                }
+                Err(err) => {
+                    debug!("{:?} Failed to receive request header: {:?}", self, err);
+                    break;
+                }
+            }
+        }
```

Comment:
Remember to purge_outdated_tasks() before recv_loop() exit

Reply:
OK, same as `gracefully shutdown` here.